scripts/bravo_csv.py

```diff
@@ -236,38 +238,347 @@ def aliquot_fixed_volume(currentStep, lims, volume, log):
 
 def prepooling(currentStep, lims):
     log = []
-    # First thing to do is to grab the volumes of the input artifacts. The method is ... rather unique.
-    data = compute_transfer_volume(currentStep, lims, log)
-    with open("bravo.csv", "w") as csvContext:
-        for s in data:
-            if s['vol_to_take'] > MAX_WARNING_VOLUME:
-                log.append("Volume for sample {} is above {}, redo the calculations manually".format(MAX_WARNING_VOLUME, s['name']))
-            if s['vol_to_take'] < MIN_WARNING_VOLUME:
-                log.append("Volume for sample {} is below {}, redo the calculations manually".format(MIN_WARNING_VOLUME, s['name']))
-            csvContext.write("{0},{1},{2},{3},{4}\n".format(s['src_fc_id'], s['src_well'], s['vol_to_take'], s['dst_fc'], s['dst_well']))
-    if log:
-        with open("bravo.log", "w") as logContext:
-            logContext.write("\n".join(log))
-
-    df = pd.read_csv("bravo.csv", header=None)
-    df['dest_row'] = df.apply(lambda row: row[4].split(':')[0], axis=1)
-    df['dest_col'] = df.apply(lambda row: int(row[4].split(':')[1]), axis=1)
-    df = df.sort_values(['dest_col', 'dest_row']).drop(['dest_row', 'dest_col'], axis=1)
-    df.to_csv('bravo.csv', header=False, index=False)
+    if currentStep.instrument.name == "Zika":
+        # Constraints
+        zika_min_vol = 0.5  # Possible to run on 0.1
+        zika_max_vol = 5
+        src_dead_vol = 5
+        pool_max_vol = 170
+
+        # Create dataframe of all transfers incl. transfer volume
+        df = zika_calc(currentStep, lims, log, zika_min_vol, src_dead_vol, pool_max_vol)
+
+        # Create worklist and log files
+        wl_filename, log_filename = zika_wl(df, zika_min_vol, zika_max_vol, src_dead_vol, pool_max_vol, log, currentStep.id)
+
+        # Couple files to LIMS
+        for out in currentStep.all_outputs():
+            if out.name == "Mosquito CSV File":
+                for f in out.files:
+                    lims.request_session.delete(f.uri)
+                lims.upload_new_file(out, wl_filename)
+            if out.name == "Mosquito Log":
+                for f in out.files:
+                    lims.request_session.delete(f.uri)
+                lims.upload_new_file(out, log_filename)
+
+        # LIMS messages
+        if any("Error:" in entry for entry in log):
+            sys.stderr.write("Critical error, please check the Log file\n")
+            sys.exit(2)
+        elif any("Warning:" in entry for entry in log):
```

Comment:
There is a mixture of "WARNING" and "Warning" in the scripts. I will suggest change all "Warning" into "WARNING".